lib/cn/csched_sp3_work.c

```diff
@@ -429,8 +419,25 @@ sp3_work_leaf_len(
             }
         }
 
-        if (runlen >= runlen_min)
-            return runlen;
+        if (runlen >= runlen_min) {
+
+            /* If scatter is low then simply kcompact the full run of kvsets.
+             */
+            if (scatter < thresh->lscatter_hwm)
+                return runlen;
+
+            /* If the run is longer than the minimum then kcompact just the
```

Comment:
Just clarifying - this means that there could exist kvsets whose vgroup count exceeds the configured vgroups max for a kvdb, right?